src/cmap/establish/handshake.rs

```diff
@@ -335,6 +335,9 @@ pub(crate) struct Handshaker {
     http_client: crate::runtime::HttpClient,
 }
 
+#[cfg(test)]
+pub(crate) static METADATA: std::sync::OnceLock<ClientMetadata> = std::sync::OnceLock::new();
```

Comment:
This gives me the heebie-jeebies but since it's explicitly test-restricted and a lot less disruptive than trying to propagate the computed metadata back up the ownership/worker hierarchy it seemed the less-bad option.